packages/runner-shared/src/index.ts

```diff
@@ -0,0 +1,41 @@
+export * from './snapshot-controls'
+
+export * from './visit-failure'
+
+export * from './blank-contents'
+
+export * from './message'
+
+export * from './selector-playground'
+
+export * from './script-error'
+
+export * from './automation-disconnected'
+
+export * from './no-automation'
+
+export * from './mobx'
+
+export * from './error-messages'
+
+export * from './iframe'
+
+export * from './dom'
+
+export * from './logger'
+
+export * from './event-manager'
+
+export * from './automation'
+
+export * from './studio'
+
+export * from './viewport-info'
+
+export * from './config-file-formatted'
+
+export * from './header'
+
+export * from './container'
+
+export * from './automation-element'
```

Comment:
I'm not sure I agree with how these separate namespaces are all jumbled together and exported, I don't understand if there is a good reason for this.

You can't do `export * as Foo from './foo'`, but you can do it in two steps, like in `packages/network/lib/index.ts`: https://github.com/cypress-io/cypress/blob/77880f1a0f0cecaee92e1f271cede452cc133287/packages/network/lib/index.ts#L1-L19

@lmiller1990 can this be updated so this so that it's namespaced by module instead of mixed up?

Reply:
This is (mostly) set up in a similar manner. There is two main scenarios to consider: 

## 1. single files (inside of `packages/runner-shared/src`). Examples are `dom.js` and `logger.js`. 

Each root level file has a single named export - for example:
- 
- Each of those exports is either a file (eg `packages/runner-shared/dom.js`) with a named export:

```js
// packages/runner-shared/src/dom.js

/* implementation */

export const dom = {
  addBlackout,
  removeBlackouts,
  addElementBoxModelLayers,
  // many more functions
}
```

So when you do `export * from './dom'`, it's just the same as what you are suggesting - except instead of import and re-exporting, we just do it in one line.

Usage is there:

```js
import { dom } from '@packages/runner-shared'`

dom.addBlackout()
```

So it is still namespaced. This same pattern is applied to all root level files.

## 2. Directories with multiple files

The other scenario is a directory, like `snapshot-controls`. It has three files:

```jsx
- index.js
- snapshot-controls.jsx
- snapshot-controls.spec.jsx
```

`index.js` simply exposes the module interface, in this case it does `export * as from ./snapshot-controls'`. It only contains 1 named export, `<SnapshotControls />` so name-spacing it doesn't make much sense. We could do `<Snapshot.Controls />`, but it seems overkill (a `Snapshot` namespace for a single component?) 

Usage is:

```
import { SnapshotControls, dom, logger } from 'packages/runner-shared`

So it's pretty well namespaced.

## 3. Exception

The only exception is the `studio` directory, which two components (`<Studio>` and `<StudioModals>`) and one singleton (`studioModel`). We could namespace these, but it feels a bit odd to have React components and a global singleton in the same namespace:

```jsx
import { Studio } from '@packages/runner-shared'

export function SomeComponent () {
  studio.studioModel(/* stuff */)
  
  return (
    <studio.StudioModals />
  )
}
}
```

What do you think?

Reply:
you have convinced me that it's not worth changing, at least :smile: